src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -1196,29 +1194,19 @@ static void mmap_attach_shared(int vmid, char** addr, size_t* sizep, TRAPS) {
               "Process not found");
   }
 
+  // open the shared memory file for the give vmid
   char* filename = get_sharedmem_filename(dirname, vmid, nspid);
 
-  // copy heap memory to resource memory. the open_sharedmem_file
-  // method below need to use the filename, but could throw an
-  // exception. using a resource array prevents the leak that
-  // would otherwise occur.
-  char* rfilename = NEW_RESOURCE_ARRAY(char, strlen(filename) + 1);
-  strcpy(rfilename, filename);
+  // Handle the exception explicitly so we can free the strings.
+  int fd = open_sharedmem_file(filename, file_flags, THREAD);
 
   // free the c heap resources that are no longer needed
   FREE_C_HEAP_ARRAY(char, luser);
   FREE_C_HEAP_ARRAY(char, dirname);
   FREE_C_HEAP_ARRAY(char, filename);
 
-  // open the shared memory file for the give vmid
-  int fd = open_sharedmem_file(rfilename, file_flags, THREAD);
-
-  if (fd == OS_ERR) {
-    return;
-  }
```

Comment:
I think you still need this for the case where the file is insecure - no exception is thrown in that case.

Reply:
Fixed.